app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,544 @@
+      public void onServiceConnected(ComponentName className, IBinder service) {
+        HotspotService.HotspotBinder binder = (HotspotService.HotspotBinder) service;
+        hotspotService = binder.getService();
+        bound = true;
+        hotspotService.registerCallBack(ZimHostActivity.this);
+      }
+
+      @Override
+      public void onServiceDisconnected(ComponentName arg0) {
+        bound = false;
+      }
+    };
+
+    serviceIntent = new Intent(this, HotspotService.class);
+
+    startServerButton.setOnClickListener(v -> {
+      //Get the path of ZIMs user has selected
+      if (!isServerStarted) {
+        getSelectedBooksPath();
+        if (selectedBooksPath.size() > 0) {
+          startHotspotHelper();
+        } else {
+          Toast.makeText(ZimHostActivity.this, R.string.no_books_selected_toast_message,
+              Toast.LENGTH_SHORT).show();
+        }
+      } else {
+        startHotspotHelper();
+      }
+    });
+  }
+
+  private void startHotspotHelper() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      toggleHotspot();
+    } else {
+      //TO DO: show Dialog() + within that add check mobile Data check later.
+      //if (isMobileDataEnabled(context)) {
+      //  mobileDataDialog();
+      //} else {
+      if (isServerStarted) {
+        startService(ACTION_STOP_SERVER);
+      } else {
+        startHotspotDialog();
+      }
+      //}
+    }
+  }
+
+  private void getSelectedBooksPath() {
+    BooksOnDiskListItem.BookOnDisk bookOnDisk;
+
+    for (BooksOnDiskListItem item : booksAdapter.getItems()) {
+      if (item.isSelected()) {
+        bookOnDisk = (BooksOnDiskListItem.BookOnDisk) item;
+        File file = bookOnDisk.getFile();
+        selectedBooksPath.add(file.getAbsolutePath());
+        Log.v(TAG, "ZIM PATH : " + file.getAbsolutePath());
+      }
+    }
+  }
+
+  private void open(@Nullable BooksOnDiskListItem.BookOnDisk bookOnDisk) {
+    File file = bookOnDisk.getFile();
+    Intent zimFile = new Intent(this, MainActivity.class);
+    zimFile.setData(Uri.fromFile(file));
+    startActivity(zimFile);
+    finish();
+  }
+
+  private void select(@NonNull BooksOnDiskListItem.BookOnDisk bookOnDisk) {
+    ArrayList<BooksOnDiskListItem> booksList = new ArrayList<>();
+    for (BooksOnDiskListItem item : booksAdapter.getItems()) {
+      if (item.equals(bookOnDisk)) {
+        if (item.isSelected()) {
+          item.setSelected(false);
+        } else {
+          item.setSelected(true);
+        }
+      }
+      booksList.add(item);
+    }
+    booksAdapter.setItems(booksList);
+  }
+
+  @Override protected void onStart() {
+    super.onStart();
+    bindService();
+  }
+
+  @Override protected void onStop() {
+    super.onStop();
+    unbindService();
+  }
+
+  private void bindService() {
+
+    bindService(serviceIntent, serviceConnection, Context.BIND_AUTO_CREATE);
+    bound = true;
+  }
+
+  private void unbindService() {
+    if (bound && !isServerStarted) {
```

Comment:
delete the bound variable and you probably don't care about IsServerStrted either, you are just disconnecting from a service.
You will have to null check `hotspotService`

Reply:
Fixed. 106d085570cdfdba2bcb168eba7faaf280344356